File: scripts/_reescape.py
import io, glob, re
setre = re.compile(r'(?<![\\A-Za-z0-9^_}])\{((?:[^{}]|\([^()]*\))*,(?:[^{}]|\([^()]*\))*)\}')
tot = files = 0
for p in glob.glob('src/constants/*.js'):
    s = io.open(p, encoding='utf-8').read()
    s2, n = setre.subn(lambda m: '\\\\{' + m.group(1) + '\\\\}', s)
    if n:
        io.open(p, 'w', encoding='utf-8').write(s2)
        tot += n
        files += 1
print('re-escaped', tot, 'in', files)

File: scripts/_reescape.js
const fs = require('fs');
const path = require('path');

const setre = /(?<![\\A-Za-z0-9^_}])\{((?:[^{}]|\([^()]*\))*,(?:[^{}]|\([^()]*\))*)\}/g;

let tot = 0, files = 0;
const dir = path.join(__dirname, '../src/constants');
for (const f of fs.readdirSync(dir).filter(f => f.endsWith('.js'))) {
  const p = path.join(dir, f);
  const s = fs.readFileSync(p, 'utf8');
  let n = 0;
  const s2 = s.replace(setre, (m, inner) => { n++; return '\\\\{' + inner + '\\\\}'; });
  if (n) {
    fs.writeFileSync(p, s2, 'utf8');
    console.log(`  ${f}: ${n}`);
    tot += n; files++;
  }
}
console.log(`Total: ${tot} in ${files} files`);
